cmd/norduser/main.go

```diff
@@ -32,6 +36,66 @@ func openLogFile(path string) (*os.File, error) {
 	return logFile, nil
 }
 
+func addAutostart(userHomeDir string) (string, error) {
+	autostartDesktopFileContents := "[Desktop Entry]" +
+		"\nName=NordVPN" +
+		"\nExec=nordvpn user" +
+		"\nTerminal=false" +
+		"\nType=Application" +
+		"\nX-GNOME-Autostart-enabled=true" +
+		"\nX-GNOME-Autostart-Delay=10" +
+		"\nX-KDE-autostart-after=panel" +
+		"\nX-MATE-Autostart-Delay=10" +
+		"\nComment=This is an autostart for NordVPN user daemon" +
+		"\nCategories=Utility;"
+
+	path := path.Join(userHomeDir, ".config", "autostart", "nordvpn.desktop")
+	if err := internal.EnsureDir(path); err != nil {
+		return "", fmt.Errorf("ensuring path: %w", err)
+	}
+
+	// #nosec G304 -- no input comes from the user
+	file, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		return "", fmt.Errorf("opening file: %w", err)
+	}
+
+	if _, err := file.Write([]byte(autostartDesktopFileContents)); err != nil {
+		return "", fmt.Errorf("writign file: %w", err)
+	}
+
+	return path, nil
+}
+
+func shouldEnableFileshare(uid uint32) (bool, error) {
+	daemonURL := fmt.Sprintf("%s://%s", internal.Proto, internal.DaemonSocket)
+
+	grpcConn, err := grpc.Dial(
+		daemonURL,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	if err != nil {
+		return false, fmt.Errorf("can't connect to daemon: %w", err)
+	}
+
+	defer func() {
+		if err := grpcConn.Close(); err != nil {
+			log.Println("failed to close grpc connection on failure")
```

Comment:
`failed to close grpc connection`, without `on failure` because this is always executed?